var/spack/repos/builtin/packages/netlib-lapack/package.py

```diff
@@ -66,9 +64,11 @@ class NetlibLapack(Package):
     provides('blas', when='~external-blas')
     provides('lapack')
 
-    depends_on('cmake', type='build')
     depends_on('blas', when='+external-blas')
     depends_on('netlib-xblas+fortran+plain_blas', when='+xblas')
+    depends_on('python@2.7:', type='test')
+
+    _building_shared = False
```

Comment:
Why off? Should be a variant?

Reply:
There is a variant 'shared'. The thing is that with netlib-lapack's CMakeLists.txt we can't have static and shared libraries at the same time. In order to get both, we need to run every phase twice. So, when `~shared`, we run the default CMakePackage's implementations of `cmake()`, `build()`, `install()`, and `check()` and get only one building directory with `-DBUILD_SHARED_LIBS:BOOL=OFF`. When `+shared`, we run the overloaded methods that run the default CMakePackage's implementations of the phases twice with different values for `self._building_shared`. Since the result of `cmake_args` and `build_directory` depend on that flag, we get two building directories: one for static libraries (with `-DBUILD_SHARED_LIBS:BOOL=OFF`) and one for shared (with `-DBUILD_SHARED_LIBS:BOOL=ON`).

Reply:
I see, thanks for explaining. Maybe add a quick not above this line how this is used?